tests/unit/semantic_release/cli/test_config.py

```diff
@@ -1,15 +1,37 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
 from unittest import mock
 
 import pytest
 import tomlkit
 
 from semantic_release.cli.config import (
+    EnvConfigVar,
     GlobalCommandLineOptions,
+    HvcsClient,
     RawConfig,
     RuntimeContext,
 )
 from semantic_release.const import DEFAULT_COMMIT_AUTHOR
 
+if TYPE_CHECKING:
+    from typing import Any
+
+
+@pytest.mark.parametrize("remote_config, expected_token", [
+    ({ "type": HvcsClient.GITHUB.value }, EnvConfigVar(env="GH_TOKEN")),
+    ({ "type": HvcsClient.GITLAB.value }, EnvConfigVar(env="GITLAB_TOKEN")),
+    ({ "type": HvcsClient.GITEA.value }, EnvConfigVar(env="GITEA_TOKEN")),
+    ({}, EnvConfigVar(env="GH_TOKEN")),  # default not provided -> means Github
+    pytest.param({ "type": "nonexistent" }, EnvConfigVar(env=""), marks=pytest.mark.xfail),
+])
```

Comment:
I think this would be better as a separate test - `xfail`-ing here could hypothetically mask some other unexpected error that only this test case would produce

Reply:
After realizing I was using xfail incorrectly, adjusted accordingly to make another test. That other test however does just fail at the common configuration validation check prior to getting to the default token state.

Reply:
That's ok - I think that gives some coverage to the code path when the remote type isn't one of the three supported ones, anyway